### PR TITLE
mixing in the ability to send an explicit error from a mock response

### PR DESCRIPTION
**Pull request by heckj**
for feedback/review:

this is done primarily In order to test error handling - which comes in useful when you're building declarative pipelines that do just that with something like Combine. A recent thread in a closed issue (https://github.com/WeTransfer/Mocker/issues/24) made me think it might be useful to submit. There aren't any tests yet against this with these changes, but if this looks useful to y'all, I would be happy to create the relevant updates

### CODE REVIEW

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Consider to place some `MARK:` lines for Sources/Mock.swift, which is over 200 lines big.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/b7c62e58234dcff8" target="_blank">Bitrise</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Mocker: Executed 18 tests, with 0 failures (0 unexpected) in 1.690 (1.754) seconds</td>
    </tr>
  </tbody>
</table>

## Mocker.framework: Coverage: 89.02
| File | Coverage ||
| --- | --- | --- |
MockingURLProtocol.swift | 86.27% | ✅
Mock.swift | 84.62% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 56c30a4d5c8556fdd8a60b77b1db3a86dee38aad
</p>

File: Sources/Mock.swift
Comment (AvdLee) [inline Sources/Mock.swift]: Wouldn't it make more sense to rename this like:

```suggestion
    private init(url: URL? = nil, ignoreQuery: Bool = false, requestError: Error? = nil, dataType: DataType, statusCode: Int, data: [HTTPMethod: Data], additionalHeaders: [String: String] = [:], fileExtensions: [String]? = nil) {
```

This allows to set explicit types of errors and allows more usecases. We could then simply verify whether this value is set and report a failure instead if so.

**Review by AvdLee — CHANGES_REQUESTED** [review]

**Comment by AvdLee** [comment]
@heckj thanks for your contribution! I'd love to adjust the code a bit, update the documentation and add some tests. If you would be able to add that, we can merge this in!

**Comment by heckj** [comment]
I can definitely do that - might be a few days, but hopefully not as long as the first wait ;-)

**Comment by heckj** [comment]
:wave: @AvdLee - I updated this PR a while ago, so I think it's just waiting on your approval for anything further.

**Review by AvdLee — APPROVED** [review]
Thanks, this looks great! Sorry for not coming back to this any earlier. I think it slipped through somehow!

**Comment by ghost** [comment]
Congratulations! :tada: This was released as part of [Release 2.2.0](https://github.com/WeTransfer/Mocker/releases/tag/2.2.0) :rocket:<p align="right"><i>Generated by <a href="https://github.com/WeTransfer/GitBuddy">GitBuddy</a></i></p>

**Pull request merged**
